talk/objectorientation/operators.tex

```diff
@@ -86,6 +86,30 @@
   \end{block}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Friend declarations}
+  \begin{block}{Concept}
+    \begin{itemize}
+      \item Functions/classes can be declared \mintinline{cpp}{friend} within a class scope
+      \item They gain access to all private members
+      \item Useful for operators such as \mintinline{cpp}{a + b}
+      \item Don't abuse this! Don't modify an object in a friend!
+    \end{itemize}
+  \end{block}
+  \begin{block}{operator+ as a friend}
+    \begin{cppcode*}{gobble=2,linenos=false}
+      struct Complex {
+        float m_real; float m_imag;
+        friend Complex operator+(Complex & a, Complex & b);
+      };
+
+      Complex operator+(Complex & a, Complex & b) {
+        return Complex(a.m_real+b.m_real,a.m_imag+b.m_imag);
+      }
```

Comment:
```suggestion
      struct Complex {
        float m_real; float m_imag;
        friend Complex operator+(Complex & a, Complex & b);
      };

      Complex operator+(Complex & a, Complex & b) {
        return Complex(a.m_real+b.m_real,a.m_imag+b.m_imag);
      }
```
I think you miss `const` on the parameters.
And the point of friendship would be stronger, if `m_real` and `m_imag` were private.

Reply:
The const doesn't fit on the slide, so I removed it. Also, at this point, we haven't explained constness yet. OK to leave it out or shrink the text?

Reply:
I'll check it private fits ...

Reply:
C++ is so interdependent, I think we cannot topsort all the content to only use features that have been previously explained. And I would like to avoid having code on the slides that people should not write in practice, like an operator with a mutable argument. So yeah, I would prefer shrinking the slide to not having `const` :)

Reply:
Manage to fit everything.